### PR TITLE
Feature/rico add ingress

### PR DESCRIPTION
**Pull request by ricogu**
- add cluster issuer, wilrdcard certificate
- add reflector

### CODE REVIEW

File: apps/argocd/overlays/core/applicationsets/reflector-applicationset.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          - PruneLast=true # Allow the ability for resource pruning to happen as a final, implicit wave of a sync operation
+        # The retry feature is available since v1.7
+        retry:
+          limit: 5 # number of failed sync attempt retries; unlimited number of attempts if less than 0
+          backoff:
+            duration: 5s # the amount to back off. Default unit is seconds, but could also be a duration (e.g. "2m", "1h")
+            factor: 2 # a factor to multiply the base duration after each failed retry
+            maxDuration: 3m # the maximum amount of time allowed for the backoff strategy
+
+      ignoreDifferences: # https://github.com/argoproj/argo-cd/issues/4276#issuecomment-908455476
```
Comment (SebastianBezold) [inline apps/argocd/overlays/core/applicationsets/reflector-applicationset.yaml:52]: I guess this is a copy-paste leftover. If so, please remove it

File: apps/tls/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+name: cluster-base
```
Comment (SebastianBezold) [inline apps/tls/Chart.yaml:2]: We should adjust this name as well

File: apps/tls/values-tls.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+issuer:
+  email: "yuanxun.gu@sap.com"
```
Comment (carslen) [inline apps/tls/values-tls.yaml:2]: do we have any more general mail address available than your personal mail address for this purpose? This seems to be wrong for me.

**Review by carslen — CHANGES_REQUESTED** [review]

File: apps/tls/values-tls.yaml (hunk 3 above)
Comment (ricogu) [reply apps/tls/values-tls.yaml:2]: an email from catena-x is the best but I am not aware of that

File: apps/argocd/overlays/core/applicationsets/reflector-applicationset.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        targetRevision: '{{ targetRevision }}'
+
+      destination:
+        name: '{{ cluster-name }}'
+        namespace: reflector
+        server: ''
+
+      # Sync policy
+      syncPolicy:
+        automated: # automated sync by default retries failed attempts 5 times with following delays between attempts ( 5s, 10s, 20s, 40s, 80s ); retry controlled using `retry` field.
```
[the remark this answers is not in the document]
Comment (ricogu) [reply apps/argocd/overlays/core/applicationsets/reflector-applicationset.yaml]: done

File: apps/tls/values-tls.yaml (hunk 3 above)
Comment (ricogu) [reply apps/tls/values-tls.yaml:2]: suggestion? @carslen

Comment (carslen) [reply apps/tls/values-tls.yaml:2]: Unless there is no better solution right now, keep the personal address until we've found a better solution.

Comment (ricogu) [reply apps/tls/values-tls.yaml:2]: sure, please approve then

**Review by carslen — APPROVED** [review]

**Pull request merged**